sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -121,11 +121,16 @@ waitUntilBuilt(KernelProgramCache &Cache,
     return State == BS_Done || State == BS_Failed;
   });
 
+  if (WithBuildState->BuildResult.get()) {
+    using BuildResult = KernelProgramCache::BuildResultT;
+    const BuildResult &Res = *WithBuildState->BuildResult.get();
+    throw ExceptionT(Res.Msg, Res.Code);
+  }
+
   RetT *Result = WithBuildState->Ptr.load();
 
   if (!Result)
-    throw ExceptionT("The other thread tried to build the program/kernel but "
-                     "did not succeed.");
+    throw ExceptionT("Build of the program/kernel did not succeed previously.");
```

Comment:
If it did not succeed previously, then we should have a log/error code here, right? The if above seems to check for it, so how we can possibly reach this?

Reply:
This is a really unexpected case. Anyway it is quite possible to reach here after some changes done to program manager. Though, these changes will lead to crash of the test I'm going to add. Thus this part will be removed.